projects/rx-form-mapper/src/lib/decorators/form.decorator.ts

```diff
@@ -1,9 +1,15 @@
 import 'reflect-metadata';
-import { modelBinder } from '../bind/model-binder';
-import { FormOpts } from './form-opts';
+import { RxValidator, RxAsyncValidator, UpdateOn } from '..';
+import { ModelBinder } from '../bind/model-binder';
+
+export interface FormOpts {
+	validators?: RxValidator | RxValidator[];
+	asyncValidators?: RxAsyncValidator | RxAsyncValidator[];
+	updateOn?: UpdateOn;
+}
 
 export function Form(opts?: FormOpts): (target: any) => void {
```

Comment:
Target può esser davvero di qualsiasi tipo?

Reply:
rispondo qui ma vale anche per il commento 

> è giusto mettere any come tipo di target? O definire un tipo custom che deve essere utilizzato?
La stessa questione anche per gli altri overloading
> https://github.com/KernelPanic92/rx-form-mapper/pull/105#discussion_r609466984

è un po' complicato:

`ClassDecorator` -> target assume come valore il type della classe quindi puoi specificare "Function" o "Type<any>" al posto di any
`PropertyDecorator` | `MethodDecorator` -> target assume il valore `{ constructor: Function }` ovvero il tipo Object

premessa:
Type<T> è l'alias di 
`export declare interface Type<T> extends Function {
    new (...args: any[]): T;
}`

il typescript è in realtà un po' spreciso su quel tipo di annotazione e se provi a specificare `{ constructor: Type<any> }` anziché `{ constructor: Function }` o `Object` ti segnala come errore di compilazione qualsiasi property che provi ad annotare con quel decorator... Anche se in realtà è corretto... 

quindi ho dovuto scegliere se
1. annotare correttamente sia `ClassDecorator` sia `PropertyDecorator` ma forzare dei cast su PropertyDecorator quando passo target al binder -> target as {constructor: Type<any>} (bruttino)
2. annotare correttamente solo CLassDecorator e lasciare any al PropertyDecorator (mi sembrava di tenere il piede in due scarpe)
3. trattare sempre target come any dando per scontato che typescript faccia comunque il suo dovere come descritto da documentazione, nei metodi del binder però specifico la firma del tipo nella maniera corretta

la 1 e la 3 se la giocavano alla pari con la sola differenza che nel caso 3 non facevo distinzioni in base al tipo di decorator... mi sembrava la soluzione più politically correct 🤣

Se la risposta/decisione ti torna poco rivalutiamolo insieme 😀

Reply:
Capisco, lascia pure così. Mi sembra una decisione con una logica (della quale non avevo conoscenza) dietro